From 02193f6df5f60326ea014286532254e414510719 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@freescale.com>
Date: Tue, 21 Apr 2015 10:57:58 +0300
Subject: [PATCH 138/452] dpaa2-eth: Adjust 'options' size

The 'options' field of various MC configuration structures has changed
from u64 to u32 as of MC firmware version 7.0.

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@freescale.com>
Change-Id: I9ba0c19fc22f745e6be6cc40862afa18fa3ac3db
Reviewed-on: http://git.am.freescale.net:8181/35579
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Stuart Yoder <stuart.yoder@freescale.com>
[Xulin: Original patch taken from FSL LS2085 SDK EAR6.0,
LS2085A-SDK-SOURCE-20160304-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/ldpaa_eth.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/ldpaa_eth.c b/drivers/staging/fsl-dpaa2/ethernet/ldpaa_eth.c
index a0e48b8..a3e5b84 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/ldpaa_eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/ldpaa_eth.c
@@ -830,7 +830,7 @@ static int ldpaa_eth_init(struct net_device *net_dev)
 	uint64_t supported = 0;
 	uint64_t not_supported = 0;
 	const struct ldpaa_eth_priv *priv = netdev_priv(net_dev);
-	uint64_t options = priv->dpni_attrs.options;
+	uint32_t options = priv->dpni_attrs.options;
 
 	/* Capabilities listing */
 	supported |= IFF_LIVE_ADDR_CHANGE | IFF_PROMISC | IFF_ALLMULTI;
@@ -976,7 +976,7 @@ static void ldpaa_eth_set_rx_mode(struct net_device *net_dev)
 	int mc_count = netdev_mc_count(net_dev);
 	uint8_t max_uc = priv->dpni_attrs.max_unicast_filters;
 	uint8_t max_mc = priv->dpni_attrs.max_multicast_filters;
-	uint64_t options = priv->dpni_attrs.options;
+	uint32_t options = priv->dpni_attrs.options;
 	uint16_t mc_token = priv->mc_token;
 	struct fsl_mc_io *mc_io = priv->mc_io;
 	int err;
-- 
1.7.5.4

